Code/GraphMol/FileParsers/XYZFileParser.cpp

```diff
@@ -0,0 +1,177 @@
+    } catch (boost::bad_lexical_cast &) {
+        std::ostringstream errout;
+        errout << "Cannot convert '" << atomLine.substr(delim2, delim3 - delim2) << "' to double on line " << line << std::endl;
+        throw FileParseException(errout.str());
+    }
+
+    try {
+        pos.y = FileParserUtils::toDouble(atomLine.substr(delim4, delim5 - delim4), false);
+    } catch (boost::bad_lexical_cast &) {
+        std::ostringstream errout;
+        errout << "Cannot convert '" << atomLine.substr(delim4, delim5 - delim4) << "' to double on line " << line << std::endl;
+        throw FileParseException(errout.str());
+    }
+    
+    try {
+        pos.z = FileParserUtils::toDouble(atomLine.substr(delim6, delim7 - delim6), false);
+    } catch (boost::bad_lexical_cast &) {
+        std::ostringstream errout;
+        errout << "Cannot convert '" << atomLine.substr(delim6, delim7 - delim6) << "' to double on line " << line << std::endl;
+        throw FileParseException(errout.str());
+    }
+    
+    // create atom
+    Atom *atom = new Atom;
+    
+    std::string symb{atomLine.substr(delim0, delim1 - delim0)};
+    if (symb.size() == 2 && symb[1] >= 'A' && symb[1] <= 'Z') {
+        symb[1] = static_cast<char>(tolower(symb[1]));
+    }
+    
+    try {
+      atom->setAtomicNum(PeriodicTable::getTable()->getAtomicNumber(symb));
+    } catch (const Invar::Invariant &e) {
+      delete atom;
+      throw FileParseException(e.what());
+    }
+    
+    return atom;
+}
+
+
+RWMol *XYZDataStreamToMol(std::istream &inStream) {
+    PRECONDITION(inStream, "no stream");
+    unsigned int numAtoms = 0;
+    
+    std::string_view num{getLine(inStream)};
+    try {
+        numAtoms = FileParserUtils::toUnsigned(num);
+    } catch (boost::bad_lexical_cast &) {
+        std::ostringstream errout;
+        errout << "Unable to recognize the number of atoms: cannot convert '" << num << "' to unsigned int on line 0" << std::endl;
+        throw FileParseException(errout.str());
+    }
+    
+    std::string comment{getLine(inStream)};
+    
+    RWMol *mol = nullptr;
+    if (numAtoms != 0) {
+        mol = new RWMol();
+        Conformer *conf = new Conformer(numAtoms);
+        if (comment != "") {
+            mol->setProp("_FileComments", comment);
+        }
+        for (unsigned int i = 0; i < numAtoms; i++) {
+            if (inStream.eof()) {
+                throw FileParseException("EOF hit while reading atoms");
+            }
+            RDGeom::Point3D pos;
+            std::string_view atomLine{getLine(inStream)};
+            Atom *atom = ParseXYZFileAtomLine(atomLine, pos, i + 2);
+            unsigned int idx = mol->addAtom(atom, false, true);
+            conf->setAtomPos(idx, pos);
+            mol->setAtomBookmark(atom, idx);
+        }
+        mol->addConformer(conf);
+    }
+    
+    while (!inStream.eof()) {
+        std::string_view extraLine{getLine(inStream)};
```

Comment:
for the same reason discussed above:
```suggestion
        std::string extraLine{getLine(inStream)};
```